File: model/session.go
package model

import (
	"time"
)

type Session struct {
	ID        string    `bson:"id" json:"id"`
	UserName  string    `bson:"user_name" json:"username"`
	Title     string    `bson:"title" json:"title"`
	CreatedAt time.Time `bson:"created_at" json:"created_at"`
	UpdatedAt time.Time `bson:"updated_at" json:"updated_at"`
}

type SessionInfo struct {
	SessionID string `json:"sessionId"`
	Title     string `json:"name"`
}

File: node-backend/src/models/Session.js
import mongoose from 'mongoose';

const SessionSchema = new mongoose.Schema(
  {
    id: { type: String, required: true, unique: true, index: true },
    user_name: { type: String, required: true, index: true },
    title: { type: String, required: true },
    attached_document_id: { type: String, default: null },
    created_at: { type: Date, default: Date.now, index: true },
    updated_at: { type: Date, default: Date.now },
  },
  { versionKey: false },
);

export default mongoose.models.Session || mongoose.model('Session', SessionSchema);
